dts/bindings/spi/st,stm32-spi-common.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2021 Fabio Baltieri
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for STM32 SPI peripherals.
+
+include: spi-controller.yaml
+
+properties:
+    reg:
+      required: true
+
+    interrupts:
+      required: true
+
+    pinctrl-0:
+      type: phandles
+      required: false
+      description: |
+        Pin configuration for SPI signals (MISO, MOSI, SCK and optional NSS).
+        We expect that the phandles will reference pinctrl nodes.
+
+        For example the SPI3 would be
+           <&spi3_sck_pc10 &spi3_miso_pc11 &spi3_mosi_pc12>;
+
+        Example with NSS Pin
+           <&spi3_sck_pc10 &spi3_miso_pc11 &spi3_mosi_pc12 &spi3_nss_pa15>;
+
+    use-subghzspi-nss:
```

Comment:
I'd prefer we introduce a dedicated spi binding in which this prop is mandatory and exclusive.
This might add some overhead in this PR but in turn simplify ifdefery in the spi driver.

Reply:
Alright sure, done. Don't think it simplified the ifdefery at all, but at least it makes the new property disappear if there are no special spi active. The Zephyr handling of multiple compatibles feels a bit brittle though... but then it's build tested so we should be good.